.github/workflows/release.yaml

```diff
@@ -0,0 +1,36 @@
+# Release workflow: Pushing a tag triggers this workflow,
+# which uses the name of the tag as the container tag/version
+# that goes into GHCR.
+name: Release
+
+# Build and publish Docker image upon pushing a tag
+on:
+  push:
+    tag:
```

Comment:
I think this needs to be "tags", with an "s". That is probably why it is currently running on each push of your PR.

Reply:
Ahhh that was it!